src/Infrastructure.Styles.Analyzer/RMSYN1103ExtensionMethodsTestedAsNormalInvocationAnalyzer.cs

```diff
@@ -0,0 +1,112 @@
+// 
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Infrastructure.Styles.Analyzer
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class RMSYN1103ExtensionMethodsTestedAsNormalInvocationSyntaxAnalyzer : DiagnosticAnalyzer
+    {
+        public const string DiagnosticId = "RMSYN1103";
+
+        internal const string Title = "Extension methods are tested using normal method invocation syntax";
+        internal const string Message = "Extension methods should be tested with normal method invocation syntax";
+
+        internal const string Description =
+            "Extension methods should be tested using normal method invocation instead of by extension";
+
+        internal const string Category = "Readability";
+
+        private static readonly DiagnosticDescriptor Rule = new DiagnosticDescriptor(
+            DiagnosticId,
+            Title,
+            Message,
+            Category,
+            DiagnosticSeverity.Warning,
+            true,
+            Description);
+
+        private static readonly List<string> TestAttributeList = new List<string> { "Test", "TestAttribute" };
+
+        public override void Initialize (AnalysisContext context)
+        {
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            context.EnableConcurrentExecution();
+
+            context.RegisterSyntaxNodeAction(AnalyzeInvocationExpression, SyntaxKind.InvocationExpression);
+        }
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } =
+            ImmutableArray.Create(Rule);
+
+        private static void AnalyzeInvocationExpression (SyntaxNodeAnalysisContext context)
+        {
+            var invocationExpression = context.Node as InvocationExpressionSyntax;
+
+            if (!IsInTestMethod(invocationExpression!)) return;
+
+            var symbolInfo = context.SemanticModel.GetSymbolInfo(invocationExpression!, context.CancellationToken);
+            if (symbolInfo.Symbol is not IMethodSymbol { IsExtensionMethod: true } methodSymbol)
+                return;
+
+            if (methodSymbol.ReducedFrom == null)
+                return;
+
+            context.ReportDiagnostic(Diagnostic.Create(Rule, context.Node.GetLocation()));
+        }
+
+        private static bool IsInTestMethod (InvocationExpressionSyntax invocationExpression)
+        {
+            if (invocationExpression.Parent is not ExpressionStatementSyntax
+                { Parent: BlockSyntax
+                    { Parent: MethodDeclarationSyntax parentMethod }
+                })
+                return false;
```

Comment:
NO GOD NO, NO

Please add braces if the if condition spans multiple lines otherwise it is difficult to tread :)